plugincontainer/container_runner.go

```diff
@@ -4,429 +4,15 @@
 package plugincontainer
 
 import (
-	"bytes"
-	"context"
 	"errors"
-	"fmt"
-	"io"
-	"os/exec"
-	"path"
-	"runtime"
-	"strconv"
-	"strings"
-
-	"github.com/docker/docker/api/types"
-	"github.com/docker/docker/api/types/container"
-	"github.com/docker/docker/api/types/filters"
-	"github.com/docker/docker/api/types/mount"
-	"github.com/docker/docker/api/types/network"
-	"github.com/docker/docker/client"
-	"github.com/docker/docker/pkg/stdcopy"
-	"github.com/hashicorp/go-hclog"
-	"github.com/hashicorp/go-plugin"
-	"github.com/hashicorp/go-plugin/runner"
 )
 
 var (
-	_ runner.Runner = (*containerRunner)(nil)
-
-	errUnsupportedOS = errors.New("plugincontainer currently only supports Linux")
+	ErrUnsupportedOS = errors.New("plugincontainer currently only supports Linux")
```

Comment:
Are there any external consumers that need to check for this sentinel error? I'd prefer to keep it unexported until there's a concrete need to export it.

Reply:
Not that I know of, but I was more thinking that if we are returning a fixed error, might as well make it easy for people to check it with `errors.Is` if they want to?

Reply:
Yeah agreed it can be a nice pattern to enable, but in this case it's easy for clients to check whether this will be the cause of the error on their own (granted, with some insider knowledge of the library version they're using). And I'm kind of hoping in the long run we can delete the error altogether by supporting TCP sockets for other OSs.